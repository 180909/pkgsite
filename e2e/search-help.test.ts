/**
 * @license
 * Copyright 2021 The Go Authors. All rights reserved.
 * Use of this source code is governed by a BSD-style
 * license that can be found in the LICENSE file.
 */

import './global-types';
import puppeteer, { Page } from 'puppeteer';

describe('Search Help', () => {
  let page: Page;

  beforeAll(async () => {
    page = await newPage();
    await page.goto(baseURL + '/search-help');
  });

  afterAll(async () => {
    await page.close();
  });

  test('accessibility tree matches snapshot', async () => {
    const a11yTree = await page.accessibility.snapshot();
    expect(a11yTree).toMatchSnapshot();
  });

  test('desktop viewport matches image snapshot', async () => {
    const image = await page.screenshot({ fullPage: true });
    expect(image).toMatchImageSnapshot();
  });

  test('mobile viewport matches image snapshot', async () => {
    await page.emulate(puppeteer.devices['Pixel 2']);
    const image = await page.screenshot({ fullPage: true });
    expect(image).toMatchImageSnapshot();
  });
});
