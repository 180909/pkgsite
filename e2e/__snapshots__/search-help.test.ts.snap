// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accessibility tree (desktop) 1`] = `
Object {
  "children": Array [
    Object {
      "name": "Black Lives Matter",
      "role": "text",
    },
    Object {
      "name": "Support the Equal Justice Initiative",
      "role": "link",
    },
    Object {
      "name": "Link to Go Homepage",
      "role": "link",
    },
    Object {
      "name": "Search for a package",
      "role": "button",
    },
    Object {
      "description": "Search for a package",
      "name": "Search for a package",
      "role": "textbox",
    },
    Object {
      "description": "Why Go",
      "name": "Why Go",
      "role": "link",
    },
    Object {
      "description": "Getting Started",
      "name": "Getting Started",
      "role": "link",
    },
    Object {
      "description": "Discover Packages",
      "name": "Discover Packages",
      "role": "link",
    },
    Object {
      "name": "About",
      "role": "link",
    },
    Object {
      "level": 1,
      "name": "Search help",
      "role": "heading",
    },
    Object {
      "name": "You can use symbols or words in your search to make your search results more precise.",
      "role": "text",
    },
    Object {
      "level": 2,
      "name": "Search for an exact match",
      "role": "heading",
    },
    Object {
      "name": "Put a word or phrase inside quotes. For example,",
      "role": "text",
    },
    Object {
      "name": "\\"go cloud\\"",
      "role": "link",
    },
    Object {
      "name": ".",
      "role": "text",
    },
    Object {
      "level": 2,
      "name": "Combine searches",
      "role": "heading",
    },
    Object {
      "name": "Put OR between each search query. For example,",
      "role": "text",
    },
    Object {
      "name": "yaml OR json",
      "role": "link",
    },
    Object {
      "name": ".",
      "role": "text",
    },
    Object {
      "level": 2,
      "name": "Search by package path",
      "role": "heading",
    },
    Object {
      "name": "You can search for a package by its full or partial import path. For example,",
      "role": "text",
    },
    Object {
      "name": "go/packages",
      "role": "link",
    },
    Object {
      "name": ".",
      "role": "text",
    },
    Object {
      "name": "If the query matches a package import path, you will be redirected to the package details page for the latest version of that package. For example,",
      "role": "text",
    },
    Object {
      "name": "golang.org/x/tools/go/packages",
      "role": "link",
    },
    Object {
      "name": ".",
      "role": "text",
    },
    Object {
      "description": "Why Go",
      "name": "Why Go",
      "role": "link",
    },
    Object {
      "description": "Use Cases",
      "name": "Use Cases",
      "role": "link",
    },
    Object {
      "description": "Case Studies",
      "name": "Case Studies",
      "role": "link",
    },
    Object {
      "description": "Getting Started",
      "name": "Getting Started",
      "role": "link",
    },
    Object {
      "name": "Playground",
      "role": "link",
    },
    Object {
      "name": "Tour",
      "role": "link",
    },
    Object {
      "name": "Stack Overflow",
      "role": "link",
    },
    Object {
      "description": "Discover Packages",
      "name": "Discover Packages",
      "role": "link",
    },
    Object {
      "description": "About",
      "name": "About",
      "role": "link",
    },
    Object {
      "name": "Download",
      "role": "link",
    },
    Object {
      "name": "Blog",
      "role": "link",
    },
    Object {
      "name": "Release Notes",
      "role": "link",
    },
    Object {
      "name": "Brand Guidelines",
      "role": "link",
    },
    Object {
      "name": "Code of Conduct",
      "role": "link",
    },
    Object {
      "description": "Connect",
      "name": "Connect",
      "role": "link",
    },
    Object {
      "name": "Twitter",
      "role": "link",
    },
    Object {
      "name": "GitHub",
      "role": "link",
    },
    Object {
      "name": "Slack",
      "role": "link",
    },
    Object {
      "name": "Meetup",
      "role": "link",
    },
    Object {
      "name": "Gopher in flight goggles",
      "role": "img",
    },
    Object {
      "name": "Copyright",
      "role": "link",
    },
    Object {
      "name": "Terms of Service",
      "role": "link",
    },
    Object {
      "name": "Privacy Policy",
      "role": "link",
    },
    Object {
      "name": "Report an Issue",
      "role": "link",
    },
    Object {
      "name": "golang.org",
      "role": "link",
    },
    Object {
      "name": "Google logo",
      "role": "link",
    },
  ],
  "name": "Search Help · pkg.go.dev",
  "role": "WebArea",
}
`;

exports[`accessibility tree (mobile) 1`] = `
Object {
  "children": Array [
    Object {
      "name": "Link to Go Homepage",
      "role": "link",
    },
    Object {
      "name": "Search for a package",
      "role": "button",
    },
    Object {
      "description": "Search for a package",
      "name": "Search for a package",
      "role": "textbox",
    },
    Object {
      "name": "Open navigation.",
      "role": "button",
    },
    Object {
      "level": 1,
      "name": "Search help",
      "role": "heading",
    },
    Object {
      "name": "You can use symbols or words in your search to make your search results more precise.",
      "role": "text",
    },
    Object {
      "level": 2,
      "name": "Search for an exact match",
      "role": "heading",
    },
    Object {
      "name": "Put a word or phrase inside quotes. For example,",
      "role": "text",
    },
    Object {
      "name": "\\"go cloud\\"",
      "role": "link",
    },
    Object {
      "name": ".",
      "role": "text",
    },
    Object {
      "level": 2,
      "name": "Combine searches",
      "role": "heading",
    },
    Object {
      "name": "Put OR between each search query. For example,",
      "role": "text",
    },
    Object {
      "name": "yaml OR json",
      "role": "link",
    },
    Object {
      "name": ".",
      "role": "text",
    },
    Object {
      "level": 2,
      "name": "Search by package path",
      "role": "heading",
    },
    Object {
      "name": "You can search for a package by its full or partial import path. For example,",
      "role": "text",
    },
    Object {
      "name": "go/packages",
      "role": "link",
    },
    Object {
      "name": ".",
      "role": "text",
    },
    Object {
      "name": "If the query matches a package import path, you will be redirected to the package details page for the latest version of that package. For example,",
      "role": "text",
    },
    Object {
      "name": "golang.org/x/tools/go/packages",
      "role": "link",
    },
    Object {
      "name": ".",
      "role": "text",
    },
    Object {
      "description": "Why Go",
      "name": "Why Go",
      "role": "link",
    },
    Object {
      "description": "Use Cases",
      "name": "Use Cases",
      "role": "link",
    },
    Object {
      "description": "Case Studies",
      "name": "Case Studies",
      "role": "link",
    },
    Object {
      "description": "Getting Started",
      "name": "Getting Started",
      "role": "link",
    },
    Object {
      "name": "Playground",
      "role": "link",
    },
    Object {
      "name": "Tour",
      "role": "link",
    },
    Object {
      "name": "Stack Overflow",
      "role": "link",
    },
    Object {
      "description": "Discover Packages",
      "name": "Discover Packages",
      "role": "link",
    },
    Object {
      "description": "About",
      "name": "About",
      "role": "link",
    },
    Object {
      "name": "Download",
      "role": "link",
    },
    Object {
      "name": "Blog",
      "role": "link",
    },
    Object {
      "name": "Release Notes",
      "role": "link",
    },
    Object {
      "name": "Brand Guidelines",
      "role": "link",
    },
    Object {
      "name": "Code of Conduct",
      "role": "link",
    },
    Object {
      "description": "Connect",
      "name": "Connect",
      "role": "link",
    },
    Object {
      "name": "Twitter",
      "role": "link",
    },
    Object {
      "name": "GitHub",
      "role": "link",
    },
    Object {
      "name": "Slack",
      "role": "link",
    },
    Object {
      "name": "Meetup",
      "role": "link",
    },
    Object {
      "name": "Gopher in flight goggles",
      "role": "img",
    },
    Object {
      "name": "Copyright",
      "role": "link",
    },
    Object {
      "name": "Terms of Service",
      "role": "link",
    },
    Object {
      "name": "Privacy Policy",
      "role": "link",
    },
    Object {
      "name": "Report an Issue",
      "role": "link",
    },
    Object {
      "name": "golang.org",
      "role": "link",
    },
    Object {
      "name": "Google logo",
      "role": "link",
    },
  ],
  "name": "Search Help · pkg.go.dev",
  "role": "WebArea",
}
`;
